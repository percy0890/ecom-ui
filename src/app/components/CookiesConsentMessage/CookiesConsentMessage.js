import React from 'react';
import PropTypes from 'prop-types';
import './CookiesConsentMessage.scss';
import Button from '../Button';

const CookiesConsentMessage = ({ isModalOpen, closeModal }) =>
  isModalOpen && (
    <div className="cookie-policy-message" role="presentation">
      <p>
        Our website uses cookies and other similar technologies to improve our
        site and your online experience. By continuing to use our website you
        consent to cookies being used.
      </p>
      <span>
        <Button
          buttonType="secondary"
          customClass="confirm-btn"
          onClick={closeModal}
          label="OK"
        />
      </span>
    </div>
  );

CookiesConsentMessage.propTypes = {
  isModalOpen: PropTypes.bool,
  closeModal: PropTypes.func,
};

export default CookiesConsentMessage;
