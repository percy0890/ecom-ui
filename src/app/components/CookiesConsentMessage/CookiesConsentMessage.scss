@import '~app/styles/core';

.cookie-policy-message {
  width: 100%;
  z-index: 5000;
  position: fixed;
  bottom: 0;
  right: 0;
  padding: 1.6rem;
  display: flex;
  box-shadow: $app-shadow-element;
  background: $app-text-secondary;

  p {
    flex: 9;
    color: $app-text-white;
    padding: 0 4.5rem;
    font-size: 1.4rem;
    line-height: 1.8rem;

    @include media(mobile down) {
      padding: 0.5rem;
    }
  }

  span {
    flex: 1;
    align-items: center;
    display: flex;
    justify-content: flex-end;

    .confirm-btn {
      padding: 0.75rem 1.5rem;
    }
  }
}