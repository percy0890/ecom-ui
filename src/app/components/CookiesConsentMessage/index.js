import CookiesConsentMessage from './CookiesConsentMessage';
export default CookiesConsentMessage;
