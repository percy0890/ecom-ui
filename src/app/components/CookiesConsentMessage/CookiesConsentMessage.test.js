import React from 'react';
import { shallow } from 'enzyme';
import CookiesConsentMessage from './CookiesConsentMessage';

describe('CookiesConsentMessage Component', () => {
  let wrapper;
  const defaultFunction = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <CookiesConsentMessage isModalOpen closeModal={defaultFunction} />,
    );
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
