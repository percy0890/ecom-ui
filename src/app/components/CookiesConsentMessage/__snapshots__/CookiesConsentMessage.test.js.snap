// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CookiesConsentMessage Component renders correctly 1`] = `
<div
  className="cookie-policy-message"
  role="presentation"
>
  <p>
    Our website uses cookies and other similar technologies to improve our site and your online experience. By continuing to use our website you consent to cookies being used.
  </p>
  <span>
    <Button
      buttonType="secondary"
      customClass="confirm-btn"
      label="OK"
      onClick={[MockFunction]}
    />
  </span>
</div>
`;
